pkg/engine/resolve/resolve.go

```diff
@@ -1075,6 +1071,8 @@ func (r *Resolver) resolveObject(ctx *Context, object *Object, data []byte, obje
 			typeName, _, _, _ := jsonparser.Get(fieldData, "__typename")
 			if !bytes.Equal(typeName, object.Fields[i].OnTypeName) {
 				typeNameSkip = true
+				// Restore the response elements that may have been reset above.
+				ctx.responseElements = responseElements
```

Comment:
I don't understand this code at all, but do you need to restore ctx.lastFetchID too?

Reply:
Indeed, it looks like that's the case.